angular/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Input,
+  OnInit,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from '@angular/forms';
+import { SliderPointerChange } from './slider-pointer.component';
+
+const CUI_SLIDER_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SliderComponent),
+  multi: true
+};
+
+export interface SliderChange {
+  high: number;
+  low: number;
+}
+
+@Component({
+  selector: 'cui-slider',
+  template: `
+    <span #sliderBar class="cui-slider__bar"></span>
+    <span class="cui-slider__selection" [ngStyle]="{left: selectionLeft, width: selectionWidth}"></span>
+    <cui-slider-pointer
+      *ngIf="isObject(value)"
+      [position]="((sliderLow - min) / (max - min)) * 100"
```

Comment:
can these calculations be in the component class as a get property?

```ts
get sliderLowPosition() {
    return ((this.sliderLow - this.min) / (max - min)) * 100
}
```

Its more unit testable that way

Reply:
updated